### ISSUE TITLE
Application.Yml properties not overridden when defined in configserver

### ISSUE THREAD

**Issue by ulmermark** [spring-cloud/spring-cloud-config#588]
We are experiencing issues when "local" properties defined in the application application.yml in no specific profile are not being overriden by values defined in the active profile in the config server.

For instance.
```
local app application.yml
myservice:
   timeout: 1
```
```
configserver/myservice/myservice.yml
spring:
   profiles:  remote

myservice:
   timeout: 2
```
When the app is run using configserver, eureka and spring.profiles.active=cloud, the value for myservice.timeout is 1, when using
```
public Widget( @Value("${myservice.timeout:3"} Integer timeout){
}
```

If I comment out the setting of the value to 1 in the application.yml, recompile and execute again...now the value is 2.

It appears that the values in the application.yml, that are not assigned to a specific profile, are never overwritten when using the remote configserver and a specific profile.

Current libraries:

buildscript {
    ext {
        springBootVersion = '1.4.0.RELEASE'
        springCloudVersion = 'Camden.SR2'
    }

**Comment by spencergibb** [comment]
if you use profile `cloud`, configserver will look for `myservice-cloud.yml` which it will not find.

**Comment by ulmermark** [comment]
Apologies...those values are just examples....the active profile should of been 'remote' to match the value of the profile defined in the configserver/myservice/myservice.yml

**Comment by spencergibb** [comment]
I don't believe config server interrogates the profiles in a yml file.

**Comment by ulmermark** [comment]
Here is the log info coming from the ConfigServer...it appears to be "finding" the proper sources of properties into the environment...

```
o.s.c.c.s.e.NativeEnvironmentRepository  : Adding property source: file:/home/ulmermark/gitrepos/spn-config-repo/spn-auth-svc/spn-auth-svc.yml#local-cloud
o.s.c.c.s.e.NativeEnvironmentRepository  : Adding property source: file:/home/ulmermark/gitrepos/spn-config-repo/application.yml#local-cloud
o.s.c.c.s.e.NativeEnvironmentRepository  : Adding property source: file:/home/ulmermark/gitrepos/spn-config-repo/spn-auth-svc/spn-auth-svc.yml
o.s.c.c.s.e.NativeEnvironmentRepository  : Adding property source: file:/home/ulmermark/gitrepos/spn-config-repo/application.yml
```

The profile being used is local-cloud,  it is finding sources for the application named spn-auth-svc in the following locations:

- spn-config-repo/spn-auth-svc/spn-auth-svc.yml#local-cloud ( profile  within a bigger application specific yml)
- spn-config-repo/application.yml#local-cloud ( profile within a bigger global yml)
- spn-config-repo/spn-auth-svc/spn-auth-svc.yml ( default profile  within a bigger application specific 
- spn-config-repo/application.yml ( default profile  within a global yml)
-

**Comment by spencergibb** [comment]
what does `/env` say

**Comment by ulmermark** [comment]
Here is /env of the actual application

```
 "applicationConfig: [classpath:/application.yml]" : {
    "server.port" : "${PORT:8080}",
    "spring.http.multipart.maxFileSize" : "128MB",
    "spring.http.multipart.maxRequestSize" : "128MB",
    "spring.jackson.serialization.INDENT_OUTPUT" : true,
    "spring.jackson.serialization.WRITE_DATES_AS_TIMESTAMPS" : false,
    "spn-pem-svc.client.url" : "http://localhost:8083",
    "spn-storage-svc.client.url" : "http://localhost:8098",
    "spn-rpps-biller-svc.client.url" : "http://localhost:8088",
    "spn-opis-svc.clients[0].searchName" : "SMART PAYMENT PLAN",

    **"spn-opis-svc.clients[0].duplicateSubmissionWindow" : -1,**

    "spn-opis-svc.clients[1].searchName" : "SMART FUNDS VERIFICATION",

    **"spn-opis-svc.clients[1].duplicateSubmissionWindow" : -1,**

    "spn-opis-svc.clients[2].searchName" : "MAINSTREET BANK",

    **"spn-opis-svc.clients[2].duplicateSubmissionWindow" : -1,**

    "hystrix.command.default.execution.isolation.strategy" : "THREAD",
    "hystrix.command.default.execution.isolation.thread.timeoutInMilliseconds" : 60000,
    "ribbon.ConnectTimeout" : 3000,
    "ribbon.ReadTimeout" : 5000,
    "logging.level.com.spn" : "INFO"
  },
  "springCloudClientHostInfo" : {
    "spring.cloud.client.hostname" : "localhost",
    "spring.cloud.client.ipAddress" : "10.0.2.15"
  },
  "spn.spring.cloud.config.runtime" : {
    "spn.spring.cloud.config.runtime.uri" : "http://localhost:8888"
  },
  "refresh" : {
    "spring.mustache.cache" : "false",
    "spring.h2.console.enabled" : "true",
    "spring.groovy.template.cache" : "false",
    "spring.mvc.log-resolved-exception" : "true",
    "spring.resources.cache-period" : "0",
    "spring.thymeleaf.cache" : "false",
    "spring.template.provider.cache" : "false",
    "spring.freemarker.cache" : "false",
    "spring.velocity.cache" : "false",
    "server.session.persistent" : "true"
  },
  "applicationConfig: [classpath:/bootstrap.yml]#local-cloud" : {
    "spring.profiles" : "local-cloud",
    "spring.cloud.config.enabled" : true,
    "spring.cloud.config.failFast" : true,
    "spring.cloud.config.discovery.enabled" : true,
    "spring.cloud.config.discovery.service-id" : "spn-config-svc",
    "eureka.client.enabled" : true,
    "eureka.client.serviceUrl.defaultZone" : "http://localhost:8761/eureka/",
    "eureka.client.serviceUrl.instanceId" : "${spring.application.name}:${spring.application.instance_id:${server.port}}"
  },
  "applicationConfig: [classpath:/bootstrap.yml]" : {
    "spring.application.name" : "spn-opis-svc",
    "spring.cloud.config.enabled" : false,
    "spring.cloud.config.retry.initial-interval" : 5000,
    "spring.cloud.config.retry.max-attempts" : 10,
    "spring.cloud.config.retry.max-interval" : 10000,
    "spring.cloud.config.retry.multiplier" : 1.1,
    "eureka.client.enabled" : false
  },
  "configService:configClient" : {
    "config.client.version" : "c523efa885b65526023402dbd7403edc344fa8f7"
  },
  "configService:file:///home/ulmermark/gitrepos/spn-config-repo/spn-opis-svc/spn-opis-svc.yml#local-cloud" : {
    "spring.profiles" : "local-cloud",
    "spn-opis-svc.clients[0].searchName" : "SMART PAYMENT PLAN",

    **"spn-opis-svc.clients[0].duplicateSubmissionWindow" : 30,**

    "spn-opis-svc.clients[1].searchName" : "SMART FUNDS VERIFICATION",

    **"spn-opis-svc.clients[1].duplicateSubmissionWindow" : 30,**

    "spn-opis-svc.clients[2].searchName" : "MAINSTREET BANK",

    _**"spn-opis-svc.clients[2].duplicateSubmissionWindow" : 30**_
  },
  "configService:file:///home/ulmermark/gitrepos/spn-config-repo/application.yml#local-cloud" : {
    "spring.profiles" : "local-cloud,local-docker"
  },
  "configService:file:///home/ulmermark/gitrepos/spn-config-repo/spn-opis-svc/spn-opis-svc.yml" : {
    "ribbon.ReadTimeout" : 90000
  },
  "configService:file:///home/ulmermark/gitrepos/spn-config-repo/application.yml" : {
    "spring.http.multipart.maxFileSize" : "256MB",
    "spring.http.multipart.maxRequestSize" : "256MB",
    "spring.cloud.config.overrideSystemProperties" : false,
    "spring.cloud.config.overrideNone" : true,
    "logging.config" : "${logback.config.base.url:http://localhost:8888}/${spring.application.name}/${spring.profiles.active}/${spring.cloud.config.label:master}/logback-spring.xml",
    "endpoints.restart.enabled" : false,
    "endpoints.restart.sensitive" : true,
    "endpoints.shutdown.enabled" : false,
    "endpoints.shutdown.sensitive" : true,
    "endpoints.trace.enabled" : false,
    "endpoints.trace.sensitive" : false,
    "endpoints.metrics.enabled" : false,
    "endpoints.metrics.sensitive" : false,
    "endpoints.routes.enabled" : false,
    "endpoints.routes.sensitive" : false,
    "endpoints.health.enabled" : true,
    "endpoints.health.sensitive" : true,
    "health.config.enabled" : false,
    "hystrix.command.default.execution.isolation.thread.timeoutInMilliseconds" : 60000,
    "ribbon.ConnectTimeout" : 3000,
    "ribbon.ReadTimeout" : 5000,
    "security.sessions" : "NEVER",
    "spn-pem-svc.client.url" : "",
    "spn-storage-svc.client.url" : "",
    "spn-rpps-biller-svc.client.url" : "",
    "spn-vendor-gateway-svc.client.url" : "",
    "spn-auth-svc.su-group-uuid" : "9a649dec-922e-11e6-ae22-56b6b6499611",
    "spn-auth-svc.su-admin" : "${spn-auth-svc.su-group-uuid}:WD_ADMIN",
    "spn-common-auth.permit-all-urls" : "/manage/**,/health",
    "spn-common-auth.su-group-uuid" : "${spn-auth-svc.su-group-uuid}"
  },
```

The particular properties I do not see being overwritten with values from the configserver are:

 "spn-opis-svc.clients[0].duplicateSubmissionWindow" : -1,
 "spn-opis-svc.clients[1].duplicateSubmissionWindow" : -1,
 "spn-opis-svc.clients[2].duplicateSubmissionWindow" : -1,

**Comment by spencergibb** [comment]
It would have been nice to have mentioned that you are using indexed properties. A couple of things. Indexed properties can not be merged across property sources. And is that reversed or is that the top to bottom order from `/env`? Looks reversed to me.

**Comment by ulmermark** [comment]
That is in the order the curl produced output.

Apologies, as I have found the issue....we added the following to our configserver the other day for the profile in question
```
    cloud:
      config:
            overrideSystemProperties: false
            overrideNone: true
```
Purely a misunderstanding on how these switches worked..  We didn't want anyway specifying a -D java property in this environment, so that we knew all properties were coming from YML files in the configserver repository.  We thought,             **_overrideSystemProperties: false_** would solve that issue, but it also keeps configserver values from being applied first in the list.
